js/main.js

```diff
@@ -119,8 +174,12 @@ function renderCards(cards, htmlLocation, replace = false) {
 async function main() {
   await defineCardComponent();
   getCSVAndMaybeRenderCollection();
-  if (CURRENT_PAGE == "/gacha.html") {
-    GACHA_BUTTON.onclick = (event) => pullAndRenderCards(CARDS_PER_PULL);
+  switch (CURRENT_PAGE) {
+    case "/gacha.html":
+      GACHA_BUTTON.onclick = (event) => pullAndRenderCards(CARDS_PER_PULL);
```

Comment:
Halp: currently the gacha button doesn't work - GACHA_BUTTON corresponds to the correct element, but the onclick doesn't seem to fire at all (pullAndRenderCards is never reached when I put a breakpoint)

Reply:
Found the issue, it's on line 78 in main.js:

`document.body.innerHTML += html;`

When it's called it recreates the entire DOM anew, which invalidates our references declared at the beginning of the file.

`const GACHA_BUTTON = document.getElementById("gacha-button");`

So what happens is that the onclick event is set, and then the DOM is reconstructed, creating an entirely new button that doesn't have the onclick event set. This can be fixed by changing line 78 to:

`document.body.insertAdjacentHTML("beforeend", html);`